modules/components/pages/inputs/pg_stream.adoc

```diff
@@ -0,0 +1,318 @@
+    batching:
+      count: 0
+      byte_size: 0
+      period: "" # No default (optional)
+      check: "" # No default (optional)
+      processors: [] # No default (optional)
+```
+
+== Metadata
+
+This input adds the following metadata fields to each message:
+
+- `mode`: Whether a message is part of the snapshot processing of existing data in a database (`snapshot`) or a data update streamed to Redpanda Connect (`streaming`).
+- `table`: The name of the database table from which the message originated.
+- `operation`: The type of database operation that generated the message, such as `INSERT`, `UPDATE` or `DELETE`.
+
+== Fields
+
+=== `dsn`
+
+The data source name (DSN) of the PostgreSQL database from which you want to stream updates. Use the format `postgres://[user[:password]@][netloc][:port][/dbname][?param1=value1&...]`.
+
+By default, PostgreSQL enforces SSL. In secure environments, you can override this mode by appending the parameter `sslmode=disable` to the connection string.
+
+*Type*: `string`
+
+```yml
+# Examples
+
+dsn: postgres://foouser:foopass@localhost:5432/foodb
+
+dsn: postgres://foouser:foopass@localhost:5432/foodb?sslmode=disable
+```
+
+=== `batch_transactions`
+
+When set to `true`, transactions are batched into a single message.
+
+include::components:partial$secret_warning.adoc[]
+
+*Type*: `bool`
+
+*Default*: `true`
+
+=== `stream_snapshot`
+
+When set to `true`, this input streams a snapshot of all existing data in the source database before streaming data changes. To use this setting, all database tables that you want to replicate _must_ have a primary key. This allows Redpanda Connect to read from multiple tables in parallel.
+
+*Type*: `bool`
+
+*Default*: `false`
+
+```yml
+# Examples
+
+stream_snapshot: true
+```
+
+=== `snapshot_memory_safety_factor`
+
+The fraction of available memory to use for streaming the database snapshot. Decimal values between `0` and `1` represent the percentage of memory to use. For example, `0.3` would allocate 30% of the available memory. Lower values make initial streaming slower, but help prevent out-of-memory errors.
+
+This option is only available when `stream_snapshot` is set to `true`.
+
+*Type*: `float`
+
+*Default*: `1`
+
+```yml
+# Examples
+
+snapshot_memory_safety_factor: 0.2
+```
+
+=== `snapshot_batch_size`
+
+The number of table rows to fetch in each batch when querying the snapshot. Leave at `0` to allow the input to determine the batch size based on `snapshot_memory_safety_factor` value.
+
+This option is only available when `stream_snapshot` is set to `true`.
+
+*Type*: `int`
+
+*Default*: `0`
+
+```yml
+# Examples
+
+snapshot_batch_size: 10000
+```
+
+=== `schema`
+
+The PostgreSQL schema from which to replicate data.
+
+*Type*: `string`
+```yml
+# Examples
+
+schema: public
+```
+
+=== `tables`
+
+A list of database table names to include in the logical replication. Specify each table name as a separate item.
+
+*Type*: `array`
+
+```yml
+
+tables:
```

Comment:
In the original docs, the formatting of the array was different. Is this correct?